common/src/main/scala/uk/ac/wellcome/elasticsearch/mappings/WorksIndex.scala

```diff
@@ -19,48 +18,60 @@ class WorksIndex @Inject()(client: HttpClient,
                            @Flag("es.type") itemType: String)
     extends Logging {
 
+  val license = objectField("license").fields(
+    keywordField("type"),
+    keywordField("licenseType"),
+    textField("label"),
+    textField("url")
+  )
+
+  val identifiers = objectField("identifiers").fields(
+    keywordField("identifierScheme"),
+    keywordField("value"),
+    keywordField("type"))
+
+  def location(fieldName: String = "locations") =
+    objectField(fieldName).fields(
+      keywordField("type"),
+      keywordField("locationType"),
+      textField("url"),
+      license
+    )
+
+  def date(fieldName: String) = objectField(fieldName).fields(
+    textField("label"),
+    keywordField("type")
+  )
```

Comment:
I think we might modify this to actually be a date at some point so i'd like to keep it separate.